From 2574449e3db5bb22daf8c7206051a084ef0affc0 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:50:39 -0400
Subject: [fs] nfsd: allow larger 4.1 session drc slots

Message-id: <1403560314-32525-98-git-send-email-bfields@redhat.com>
Patchwork-id: 84578
O-Subject: [Patch RHEL7 097/172] nfsd4: allow larger 4.1 session drc slots
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

The client is actually asking for 2532 bytes.  I suspect that's a
mistake.  But maybe we can allow some more.  In theory lock needs more
if it might return a maximum-length lockowner in the denied case.

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 24906f32378288d74289405231ddbb7120317691
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/state.h b/fs/nfsd/state.h
index fda9ce2..374c662 100644
--- a/fs/nfsd/state.h
+++ b/fs/nfsd/state.h
@@ -122,7 +122,7 @@ static inline struct nfs4_delegation *delegstateid(struct nfs4_stid *s)
 /* Maximum number of operations per session compound */
 #define NFSD_MAX_OPS_PER_COMPOUND 16
 /* Maximum  session per slot cache size */
-#define NFSD_SLOT_CACHE_SIZE  1024
+#define NFSD_SLOT_CACHE_SIZE  2048
 /* Maximum number of NFSD_SLOT_CACHE_SIZE slots per session */
 #define NFSD_CACHE_SIZE_SLOTS_PER_SESSION 32
 #define NFSD_MAX_MEM_PER_SESSION  \
-- 
1.7.1